From 1c88d26d1d2a7472e6111a4b486b2e8fcbca071b Mon Sep 17 00:00:00 2001
From: Marcin Niestroj <m.niestroj@emb.dev>
Date: Wed, 3 Apr 2024 14:13:32 +0200
Subject: [PATCH] drivers: net: nsos: fix compatibility with CONFIG_POSIX_API=n

Improve code consistency and fix CONFIG_POSIX_API=n compatibility with use
of ZSOCK_* and DNS_* macros.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>
---
 drivers/net/nsos_sockets.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/nsos_sockets.c b/drivers/net/nsos_sockets.c
index b70940697a9..98d6c429906 100644
--- a/drivers/net/nsos_sockets.c
+++ b/drivers/net/nsos_sockets.c
@@ -678,7 +678,7 @@ static int nsos_recvfrom_with_poll(struct nsos_socket *sock, void *buf, size_t l
 	int sock_flags;
 	bool non_blocking;
 
-	if (flags & MSG_DONTWAIT) {
+	if (flags & ZSOCK_MSG_DONTWAIT) {
 		non_blocking = true;
 	} else {
 		sock_flags = nsos_adapt_fcntl_getfl(sock->pollfd.fd);
@@ -890,7 +890,7 @@ static int nsos_getaddrinfo(const char *node, const char *service,
 	ret = addrinfo_from_nsos_mid(res_mid, res);
 	if (ret < 0) {
 		errno = -ret;
-		return EAI_SYSTEM;
+		return DNS_EAI_SYSTEM;
 	}
 
 	return ret;
-- 
2.45.2

